search/search_test.go

```diff
@@ -109,14 +110,130 @@ func TestSearchUnit(t *testing.T) {
 			So(err, ShouldEqual, searchErr)
 			So(m, ShouldBeNil)
 		})
+
+		Convey("test Dimension uses default search limit when none set", func() {
+
+			mockClient := &rchttp.ClienterMock{
+				DoFunc: func(ctx context.Context, req *http.Request) (*http.Response, error) {
+					return &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(searchResp))),
+					}, nil
+				},
+			}
+
+			searchCli := &Client{
+				cli: mockClient,
+				url: "http://localhost:22000",
+			}
+
+			Convey("when search is called", func() {
+				m, err := searchCli.Dimension(ctx, "12345", "time-series", "1", "geography", "Newport", Config{Offset: &offset})
+
+				Convey("then the default limit is used", func() {
```

Comment:
I would consider changing this to something like:
```
"then the request is sent with the default limit"
```
Just makes it clearer to anyone reading it exactly what we are testing/expecting